lib/logstorage/storage_search.go

```diff
@@ -587,6 +587,88 @@ func (s *Storage) GetStreamIDs(qctx *QueryContext, limit uint64) ([]ValueWithHit
 	return s.GetFieldValues(qctx, "_stream_id", limit)
 }
 
+// GetTenantIDs returns tenantIDs for the given start and end.
+func (s *Storage) GetTenantIDs(ctx context.Context, start, end int64) ([]TenantID, error) {
+	return s.getTenantIDs(ctx, start, end)
+}
+
+func (s *Storage) getTenantIDs(ctx context.Context, start, end int64) ([]TenantID, error) {
+	workersCount := cgroup.AvailableCPUs()
+	stopCh := ctx.Done()
+
+	tenantIDByWorker := make([][]TenantID, workersCount)
+	maxTenantCount := 0
+
+	// Spin up workers
+	var wgWorkers sync.WaitGroup
+	workCh := make(chan *partition, workersCount)
+	wgWorkers.Add(workersCount)
```

Comment:
It is better from the maintainability PoV to move this line as `wgWorkers.Add(1)` just before the `go func(...)` below in the loop. This will reduce chances of bugs if some additional conditions are added in front of the `go func(...)` such as `if something { continue }`. This also will make this code consistent with the rest of the codebase of VictoriaLogs, which uses the same technique.